internal/store/store.go

```diff
@@ -0,0 +1,92 @@
+package store
+
+import (
+	"database/sql"
+	"database/sql/driver"
+	"encoding/json"
+	"fmt"
+	"time"
+
+	// Register lib/pq PostreSQL driver
+	_ "github.com/lib/pq"
+)
+
+// Service is an interface to manipulate the data store.
+type Service struct {
+	db *sql.DB
+}
+
+// Options holds information for connecting to a PostgreSQL instance.
+type Options struct {
+	User, Pass string
+	Host       string
+	Port       int
+	DBName     string
+	SSLMode    string
+}
+
+// ConnectionInfo returns the PostgreSQL connection string from an options struct.
+func (o Options) ConnectionInfo() string {
+	return fmt.Sprintf("host='%s' port='%d' user='%s' password='%s' dbname='%s' sslmode='%s'", o.Host, o.Port, o.User, o.Pass, o.DBName, o.SSLMode)
+}
+
+// New creates a new store service.
+func New(o Options) (*Service, error) {
```

Comment:
This shouldn't return a pointer. Especially because it just instantly gets dereferenced when you pass it to `server.New`.